### DISCUSSION TITLE
Is it possible to remove the gap from the rightmost border of the widget?

### DISCUSSION THREAD

**Discussion by 0mega3** [jtdaugherty/brick#465] [Q&A] [answered]
Hello all,
Is it possible to remove the gap from the rightmost border of the widget?
Such gaps also exist on demo examples.
![2023-05-07_scrot](https://user-images.githubusercontent.com/46750770/236700063-e436fbc9-922b-42e2-84d3-b3bf59c35ddc.png)

**Comment by jtdaugherty** [comment]
I'd need to see more of the screenshot as well as the code itself, if you can share it. If you can't share your code, it would help if you could post a screenshot of a specific demo program so I have a reference.

**Reply by 0mega3** [reply]
Thanks for the quick answer.
I recently started studying Brick.
I used this ui ()
+  86 ui = C.vBox [C.vBox [C.fill ' '],
+  87              C.vBox [C.padBottom (C.Pad 1) $ C.vLimit 1 (C.withDefAttr introAttr (C.hLimitPercent 100 $ C.fill ' '))]]
If you run this example
https://github.com/jtdaugherty/brick/blob/master/programs/BorderDemo.hs
then the borders on it have the same gaps.

**Reply by 0mega3** [reply]
Gaps on the right side only.

**Reply by 0mega3** [reply]
![2023-05-07-232818_1920x1080_xxx_scrot](https://user-images.githubusercontent.com/46750770/236701510-6cc675ad-d4f5-46b9-9e73-d5ba05201b6e.jpg)

**Reply by jtdaugherty** [reply, marked as answer]
Oh, I see. Thanks for the screenshot, that helps! The gap you are seeing is actually because your terminal emulator puts a small gap in between the rightmost side of the rightmost column and the edge of the window. Many terminal emulators do this. The size of the gap will often depend on the font size relative to the window size. There's nothing that Brick-based applications can do about that, though.

**Reply by jtdaugherty** [reply]
(In fact, a similar gap exists in that same screenshot at the *bottom* of the image.)

**Reply by 0mega3** [reply]
Many thanks.
I use vim, but I've never paid attention to it before. As I started building the interface, I noticed this).
Brick/Vty is great. I really like. I will study. Thanks.
BR, Denis

**Reply by jtdaugherty** [reply]
I'm glad to hear that you are enjoying using the libraries!
